### PR TITLE
Add workflow to check patches

### PR DESCRIPTION
**Pull request by maarcingebala**
The action fetches code from the PR's source branch, takes the target branch, and tries to fetch and apply patches using PR's target branch. Example: if a PR targets a version branch `3.20`, the action would checkout to `3.20` branch in the patches repo and try to apply patches from that branch. If patches apply, the action succeeds, otherwise it returns an error, and indicates that patches need to be adjusted.

Other assumptions:
- Action runs on all branches.
- Action runs only when the label is applied: `check patches` (will be changed to all PRs when the workflow is more tested).

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases
- [ ] All new fields/inputs/mutations have proper labels added (`ADDED_IN_X`, `PREVIEW_FEATURE`, etc.)
- [ ] All migrations have proper dependencies
- [ ] All indexes are added concurrently in migrations
- [ ] All RunSql and RunPython migrations have revert option defined

### CODE REVIEW

File: .github/workflows/check-patches.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#
+# - Action runs on all branches.
+# - Action runs only when label is applied: `check patches` (will be changed to all PRs
+# when workflow is more tested).
+
+name: Test Patches
+
+on:
+  pull_request:
+    types: [labeled, opened, synchronize, unlabeled]
```
Comment (tomaszszymanski129) [inline .github/workflows/check-patches.yml:16]: `unlabeled` seems unnecessary?

File: .github/workflows/check-patches.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          repository: saleor/saleor-multitenant
+          token: ${{ secrets.GITHUB_TOKEN }}
+          path: saleor-multitenant
+          ref: ${{ github.base_ref }}
+
+      - name: Apply patches
+        id: apply_patches
+        run: |
+          cd workflows-test
+          git apply ../saleor-multitenant/patches/**.patch
```
Comment (tomaszszymanski129) [inline .github/workflows/check-patches.yml:42]: perhaps we wanna skip the output? something like `> /dev/null 2>&1` should do it

File: .github/workflows/check-patches.yml (hunk 1 above)
Comment (maarcingebala) [reply .github/workflows/check-patches.yml:16]: Original version I wrote used a label to skip the check, so it's a leftover from that solution. Now it's not needed indeed.

File: .github/workflows/check-patches.yml (hunk 2 above)
Comment (maarcingebala) [reply .github/workflows/check-patches.yml]: Good idea, I made the changes and the output would now look like this:
image

**Review by NyanKiyoshi — CHANGES_REQUESTED** [review]
Sent a comment at https://saleorcommerce.slack.com/archives/C06DNJPFDJB/p1726227492247399

File: .github/workflows/check-patches.yml (hunk 1 above)
Comment (maarcingebala) [inline .github/workflows/check-patches.yml:16]:
```suggestion
    types: [labeled, opened, synchronize]
```

**Review by tomaszszymanski129 — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Pull request merged**